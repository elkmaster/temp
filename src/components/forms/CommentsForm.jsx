import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Textarea from './elements/Textarea';

export const CommentsForm = ({ handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <Field
      name="comment"
      component={Textarea}
      placeholder="Enter some comment"
    />
    <button type="submit">Add comment</button>
  </form>
);

CommentsForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

const validate = ({ comment }) => {
  const errors = {};
  if (!comment) {
    errors.comment = 'This field is required';
  }
  return errors;
};

export default reduxForm({
  form: 'commentsForm', // a unique identifier for this form
  validate, // <--- validation function given to redux-form
  touchOnBlur: false,
})(CommentsForm);
