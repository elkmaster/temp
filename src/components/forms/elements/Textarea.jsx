import React from 'react';
import PropTypes from 'prop-types';

import '../../../styles/components/forms.css';

const Textarea = ({
  input,
  meta: { touched, error },
  ...rest
}) => (
  <div>
    <textarea {...input} {...rest} />
    <div className="input-invalid">
      {touched && error}
    </div>
  </div>
);

Textarea.propTypes = {
  meta: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
};

export default Textarea;
