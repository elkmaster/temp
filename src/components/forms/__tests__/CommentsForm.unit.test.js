import React from 'react';
import { shallow,  } from 'enzyme';
import faker from 'faker';

import { CommentsForm } from '../CommentsForm';
import Textarea from '../elements/Textarea';

const setup = (propsOverrides) => {
  const props = Object.assign({
      handleSubmit: jest.fn(),
    },
    propsOverrides);

  const wrapper = shallow(<CommentsForm {...props} />);

  return {
    props,
    wrapper,
  };
};

describe('CommentsForm', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should has correct input component', () => {
    const { wrapper } = setup();
    expect(wrapper.find('Field').prop('component')).toBe(Textarea);
  });

  describe('form submit', () => {
    it('should call handleSubmit on submit', () => {
      const { wrapper, props } = setup();
      wrapper.simulate('submit');
      expect(props.handleSubmit).toBeCalled();
    });

    it('should call handleSubmit with correct params', () => {
      const { wrapper, props } = setup();
      const payload = {
        values: {
          comment: faker.lorem.sentence(),
        },
      };
      wrapper.simulate('submit', payload);

      expect(props.handleSubmit).toBeCalledWith(payload);
    });
  });

});
