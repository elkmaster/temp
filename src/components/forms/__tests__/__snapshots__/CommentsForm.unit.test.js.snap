// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    name="comment"
    placeholder="Enter some comment"
  />
  <button
    type="submit"
  >
    Add comment
  </button>
</form>
`;
